src/main/java/com/github/princesslana/eriscasper/repository/GuildRepository.java

```diff
@@ -0,0 +1,14 @@
+package com.github.princesslana.eriscasper.repository;
+
+import com.github.princesslana.eriscasper.data.Snowflake;
+import com.github.princesslana.eriscasper.data.resource.Channel;
+import com.github.princesslana.eriscasper.data.resource.Guild;
+import io.reactivex.Maybe;
+import io.reactivex.annotations.NonNull;
+import java.util.Optional;
+
+public interface GuildRepository {
+  Maybe<Guild> getGuild(Optional<Snowflake> id);
```

Comment:
In what cases do we expect a client to call this when they don't have an id?

Reply:
You're right, I'll change it to a nonnull param